devito/symbolics/printer.py

```diff
@@ -94,6 +96,23 @@ def _print_Mod(self, expr):
         result = '%'.join(args)
         return result
 
+    def _print_Abs(self, expr):
+        """Print an absolute value. Use `abs` if can infer it is an Integer"""
+        # `abs` undefined in soem offloading cases
+        if isinstance(configuration['platform'], Device):
+            return "fabs(%s)" % self._print(expr.args[0])
+
+        # Check if argument is an integer
+        is_integer = True
+        for a in expr.args[0].args:
```

Comment:
should you rather use `atoms` or something like that? otherwise you might miss nested symbols if a deep expression

(test to be added accordingly if so)

Reply:
There is probably room for improvement, this is a "basic" one to remove the most obvious ones